### DISCUSSION TITLE
Hires Fix, I'm confused?

### DISCUSSION THREAD

**Discussion by deumiij** [Acly/krita-ai-diffusion#560] [Q&A] [answered]
Hello, and great work! This plugin works more than what I was expecting. Also, I hope everyone is doing fine.

1. I wanted to ask about hires fix since i can't find anything to set like in comfyui, is it automatically? I have tried using upscaler, it gets more detailed but still blurry, and i was looking for a way to tweak it? or am i doing it wrong?

something like this?
![image](https://github.com/Acly/krita-ai-diffusion/assets/108170797/4cc78621-d330-46ed-8e0f-5f2faa10f680)

I work from a small canvas and upscale it 2x while also using the model. The result is a bit blurry and has some artifacts. I wanted to add more details, despite upscaling it to even higher resolution.

![3](https://github.com/Acly/krita-ai-diffusion/assets/108170797/b570cb01-b273-42fd-88a5-a35db0b52f90)

3. I saw this in server version, is this better for quality? How do I install this on my own local comfy? Any link?
![image](https://github.com/Acly/krita-ai-diffusion/assets/108170797/e8a0d667-1bf9-4972-8542-6c6fb5a6e043)

Anyway, thank you for such plugin.

**Comment by ETCore7** [comment]
I don't use the Comfy WebUI so I don't know the answer to your 1st question.

As for upscaling manually, I didn't see blur or artifacts, I upscaled with one of the things you mentioned under 3. In the interface I have 4 options for upscaling. I never saw the interface without the optional components so I don't know whether there is a "low quality" upscaler available when you don't install that HAT and Real HAT GAN.

As for your 3rd question, when you let the plugin install everything all content can be found under
**_C:\Users\..\AppData\Roaming\krita\ai_diffusion\server\ComfyUI\models_** , so you could download it via the plugin and then move it to your own installation.

**Reply by deumiij** [reply]
Thanks for the answer, I haven't use Webui or SD in general like months so I also have no idea, maybe I'm using a bad model or something needs to be updated since I only updated comfyui as i remember.

**Comment by freecoderwaifu** [comment]
All highres fix is:

Base image gets generated at a smaller resolution (512x512 by default for SD 1.5) > base image gets upscaled using the selected upscaler to the target resolution (let's say 2x so 1024x1024) > upscaled image gets run through img2img in the background to refine it > result

You can replicate this in Krita Diffusion using the "Upscale" mode and enabling "Refine upscaled image".  The strength slider is effectively denoising strength. 

Or manually upscale an image using Krita's built in Image > Scale Image to New Size, then in the default "Generate" mode lower the strength slider below 100% and the button should change to "Refine" and then use that.

**Reply by deumiij** [reply]
Oh I see, maybe I'll try the manual way. Thank you.

**Comment by Acly** [comment, marked as answer]
Hires fix is done automatically. Just generate at any resolution, it will use two-pass generation if the resolution is outside of the checkpoint's recommended range.

The "Upscale" tab is useful for very high resolutions (it processes the image in tiles) or if you want to select a custom upscale model.
You can download HAT models from here: https://huggingface.co/Acly/hat/tree/main

**Reply by deumiij** [reply]
Thanks for the answer, I'll try to use the hat model too

**Reply by ricperry** [reply]
@Acly Thanks for this response. I don't know if you can clarify, but when I upscale beyone my vram capacity for the refiner, you're saying the upscale will compute the best tiled settings for the refiner (I think). Can we add noise at this stage and have it add more details? I think the default refiner denoise is 30%, which sometimes changes my images in weird ways, but doesn't seem to be adding detail. I usually leave this unselected because the result is usually not what I'm trying to achieve. But then my upscaled images seem to be.... lacking??? I don't know the right description. But I wonder if there's a way to add detail without changing the existing details into weird stuff. 

Also, does the upscaler/refiner use the Tiled controlnet to maintain adherence to the original image?
